### ISSUE TITLE
Advanced gear randomization unable to use basic gear randomization definitions

### ISSUE THREAD

**Issue by Drofseh** [dklollol/Olsen-Framework-Arma-3#74]
TL:DR - Allow for basic randomization to be used inside of advanced randomization.

I am trying to use advanced randomization to call a basic randomization definition. (Example 1)
See examples here: http://pastebin.com/y3FB85vN

These definitions work fine when called this way (Examples line 13, Example2)

But give a missing ] error when called as part of an advanced randomization like this (Examples line 26, Example 3, and picture)
https://i.gyazo.com/3b7fad1ed91f6555ea4b0b4b206de137.png

This means that rather than having a single definition for rifles of the same family and a single definition for all magazines that work with those rifles, each object needs its own definition, each loadout and magazine combination needs its own section in the advanced randomization, your gear sqf is several times larger than it needs to be, and you are unable to randomize magazines in the same way as with the basic randomization.

As a result, using the same objects as in the previous examples, we get this 62 line, 1625 character behemoth where each magazine type must be preselected. (Examples line 46, Example 4)

Rather than this much more compact, but not currently functional, randomization, which is only 27 lines and 813 characters, and randomizes magazines while keeping them with the appropriate rifle. (Examples line 111, Example 5)

I would like to see the method used in Example 5, or something like it, made possible.

**Comment by BlackHawkPL** [comment]
Currently only solution to your problem is to randomize parts of gearscript yourself.
This is pretty complex task, only person that can edit gearscript without breaking it is Olsen, so it's only up to him if he wants to do this.

**Comment by Drofseh** [comment]
Yah, the loadouts I'm making are definitely pushing the boundaries of what the gearscript is capable of.

**Comment by Drofseh** [comment]
Another problem I had was getting there to be a chance of nothing for a random item.

If you just you double quotes like this
```
#define I_backpack "","backpack_class1","backpack_class2"

[I_backpack] call FNC_AddItemRandom;
```
it breaks.

I was able to get around it by using brackets around everything like this
```
#define I_backpack [],["backpack_class1"],["backpack_class2"]

[I_backpack] call FNC_AddItemRandom;
```
it works, and extra [] cal be added like this
```
[],[],[I_backpack] call FNC_AddItemRandom;
```
to adjust the chance of nothing on a per case basis.

**Comment by gibfender** [comment]
It would be great to have a cleaner way of choosing nothing from a randomised list
